src/Util/ObjectUtil/ObjectUtil.js

```diff
@@ -11,6 +11,43 @@ import Logger from '../Logger';
  */
 class ObjectUtil {
 
+  /**
+   * Returns the dot delimited path of a given object by the given
+   * key-value pair. Example:
+   *
+   * ```
+   * const obj = {
+   *   level: 'first',
+   *   nested: {
+   *     level: 'second'
+   *   }
+   * };
+   * const key = 'level';
+   * const value = 'second';
+   *
+   * ObjectUtil.getPathByKeyValue(obj, key, value); // 'nested.level'
+   * ```
+   *
+   * Note: It will return the first available match!
+   *
+   * @param {Object} obj The object to obtain the path from.
+   * @param {String} key The key to look for.
+   * @param {String|Number|Boolean} value The value to look for.
+   * @param {String} [currentPath=''] The currentPath (if called in a recursion)
+   *                                  or the custom root path (default is to '').
+   */
+  static getPathByKeyValue(obj, key, value, currentPath = '') {
+    currentPath = currentPath ? `${currentPath}.` : currentPath;
+
+    for (let k in obj) {
+      if (k === key && obj[k] === value) {
+        return `${currentPath}${k}`;
+      } else if (typeof obj[k] === 'object') {
+        return ObjectUtil.getPathByKeyValue(obj[k], key, value, `${currentPath}${k}`);
```

Comment:
How do we ensure this function does not run in an infinite loop, for objects that aren't serializable, because of cyclic references?

Reply:
Not sure about this. At the moment I would say, that we support plain objects only…

Reply:
Maybe we just ad a line to the docs for now